src/python/gudhi/representations/metrics.py

```diff
@@ -46,13 +46,13 @@ def _compute_persistence_diagram_projections(X, num_directions):
     This is a function for projecting the points of a list of persistence diagrams (as well as their diagonal projections) onto a fixed number of lines sampled uniformly on [-pi/2, pi/2]. This function can be used as a preprocessing step in order to speed up the running time for computing all pairwise sliced Wasserstein distances / kernel values on a list of persistence diagrams. 
 
     Parameters:
-        X (list of n numpy arrays of shape (numx2)): list of persistence diagrams. 
+        X (list of n numpy arrays of shape (num,2)): list of persistence diagrams.
         num_directions (int): number of lines evenly sampled from [-pi/2,pi/2] in order to approximate and speed up the distance computation.
 
     Returns: 
-        list of n numpy arrays of shape (2*numx2): list of projected persistence diagrams.
+        list of n numpy arrays of shape (2*num,num_directions): list of projected persistence diagrams.
     """
-    thetas = np.linspace(-np.pi/2, np.pi/2, num=num_directions+1)[np.newaxis,:-1]
+    thetas = np.linspace(-np.pi/2, np.pi/2, num=num_directions, endpoint=False)[np.newaxis,:-1]
```

Comment:
There change is leading to an issue in CI (also on my side) in test_representations.py, line 104:
```
_______________________________________________________________ test_kernel_distance_consistency _______________________________________________________________

    def test_kernel_distance_consistency():
        l1, l2 = _n_diags(9), _n_diags(11)
        for kernelName, (kernelClass, kernelParams, tolerance) in kernel_dict.items():
            _ = kernelClass.fit(l1)
            f2 = kernelClass.transform(l2)
            f12 = np.array([[kernelClass(l1_, l2_) for l1_ in l1] for l2_ in l2])
>           assert f12 == pytest.approx(f2, **tolerance)
E           assert array([[0.56618955, 0.48974822, 0.31553973, 0.51620905, 0.49924361,\n        0.4346309 , 0.558053  , 0.49428306, 0.45360757],\n       [0.40089888, 0.44892065, 0.32447588, 0.47145567, 0.44408282,\n        0.35050003, 0.44284732, 0.43704641, 0.45059925],\n       [0.47279853, 0.47122498, 0.29967823, 0.38130872, 0.48611073,\n        0.39338417, 0.48564242, 0.42208918, 0.40996905],\n       [0.54483224, 0.48775067, 0.25332228, 0.46831764, 0.4632743 ,\n        0.37386518, 0.48512388, 0.41554467, 0.38072268],\n       [0.42595557, 0.36847709, 0.44108813, 0.36694191, 0.52528717,\n        0.51321019, 0.46892082, 0.50650572, 0.44629377],\n       [0.47735513, 0.46897831, 0.32043588, 0.44081062, 0.50076268,\n        0.36993172, 0.48789487, 0.50458098, 0.47863596],\n       [0.49088619, 0.32425983, 0.25161602, 0.31060926, 0.32683951,\n        0.37904493, 0.43892888, 0.31626628, 0.29035103],\n       [0.28716591, 0.28664174, 0.36327047, 0.24486394, 0.32937239,\n        0.32814484, 0.34740431, 0.31610328, 0.3850524 ],\n       [0.49546605, 0.3909431 , 0.36189466, 0.3854066 , 0.4884735 ,\n        0.47326653, 0.5097694 , 0.39110274, 0.38815714],\n       [0.35266545, 0.32681397, 0.41884708, 0.275671  , 0.40147208,\n        0.39980974, 0.4179076 , 0.38152168, 0.45740623],\n       [0.59400539, 0.51948161, 0.30762344, 0.46649581, 0.49927177,\n        0.42322176, 0.53957685, 0.51443973, 0.46702   ]]) == approx([[0.5749909668197436 ± 5.7e-04, 0.4982809318565699 ± 5.0e-04, 0.3134150250073149 ± 3.1e-04, 0.5207541150644102 ± 5.2e-04, 0.5113092316865084 ± 5.1e-04, 0.44782135221718467 ± 4.5e-04, 0.5638358511267394 ± 5.6e-04, 0.5087362501516259 ± 5.1e-04, 0.4583123800154211 ± 4.6e-04], [0.40391136094126817 ± 4.0e-04, 0.4629188577066109 ± 4.6e-04, 0.3270217988289004 ± 3.3e-04, 0.4744905988910448 ± 4.7e-04, 0.45122069617141375 ± 4.5e-04, 0.36473938291538227 ± 3.6e-04, 0.45076334272894086 ± 4.5e-04, 0.45627088548274264 ± 4.6e-04, 0.45724005049219124 ± 4.6e-04], [0.49234843880437607 ± 4.9e-04, 0.4837867784820462 ± 4.8e-04, 0.30479843354026726 ± 3.0e-04, 0.3877904779637883 ± 3.9e-04, 0.5091315269498977 ± 5.1e-04, 0.41258599322508044 ± 4.1e-04, 0.5073916624412993 ± 5.1e-04, 0.4335772634543771 ± 4.3e-04, 0.41889222577946744 ± 4.2e-04], [0.5581575375417775 ± 5.6e-04, 0.4956813043068013 ± 5.0e-04, 0.24939335696155615 ± 2.5e-04, 0.4705759379193394 ± 4.7e-04, 0.4675657676343013 ± 4.7e-04, 0.38311515208298363 ± 3.8e-04, 0.49676797773027376 ± 5.0e-04, 0.4205948863686782 ± 4.2e-04, 0.38195189904435023 ± 3.8e-04], [0.4447724991306245 ± 4.4e-04, 0.3783054208340637 ± 3.8e-04, 0.4390701786134871 ± 4.4e-04, 0.36846865244987703 ± 3.7e-04, 0.5270471107170016 ± 5.3e-04, 0.5245109429718812 ± 5.2e-04, 0.48478513819496744 ± 4.8e-04, 0.5166881255308146 ± 5.2e-04, 0.4603307499137023 ± 4.6e-04], [0.48662240966122744 ± 4.9e-04, 0.48060577303582214 ± 4.8e-04, 0.318920320846488 ± 3.2e-04, 0.4402181698608169 ± 4.4e-04, 0.5065287479959606 ± 5.1e-04, 0.3808428016344234 ± 3.8e-04, 0.503760518497169 ± 5.0e-04, 0.5119492939077402 ± 5.1e-04, 0.47889476224616057 ± 4.8e-04], [0.4916554733557071 ± 4.9e-04, 0.32723491697559476 ± 3.3e-04, 0.2503988728840708 ± 2.5e-04, 0.3060085611576825 ± 3.1e-04, 0.33358102847477644 ± 3.3e-04, 0.38624049162552265 ± 3.9e-04, 0.4409724167911789 ± 4.4e-04, 0.3228061546269177 ± 3.2e-04, 0.2965345226568545 ± 3.0e-04], [0.2845018638550871 ± 2.8e-04, 0.28188355320070757 ± 2.8e-04, 0.3746961222903527 ± 3.7e-04, 0.239383289320872 ± 2.4e-04, 0.3380651566374858 ± 3.4e-04, 0.33980877688422756 ± 3.4e-04, 0.34734524655111854 ± 3.5e-04, 0.32213977069801136 ± 3.2e-04, 0.3963803850825761 ± 4.0e-04], [0.5094956203659733 ± 5.1e-04, 0.40490180684472027 ± 4.0e-04, 0.3638743592115522 ± 3.6e-04, 0.3839958751874655 ± 3.8e-04, 0.5032914529263811 ± 5.0e-04, 0.4885345815871985 ± 4.9e-04, 0.5174383521666145 ± 5.2e-04, 0.40837256558614843 ± 4.1e-04, 0.40339167731136283 ± 4.0e-04], [0.3510676862025384 ± 3.5e-04, 0.3244424797642443 ± 3.2e-04, 0.4328676030379516 ± 4.3e-04, 0.27242857987215746 ± 2.7e-04, 0.42216604675489317 ± 4.2e-04, 0.41882884230174644 ± 4.2e-04, 0.4169958992359919 ± 4.2e-04, 0.3957578636483203 ± 4.0e-04, 0.4698541251673909 ± 4.7e-04], [0.6039527453563808 ± 6.0e-04, 0.5220245309404515 ± 5.2e-04, 0.30428597130705143 ± 3.0e-04, 0.4691870470253766 ± 4.7e-04, 0.5168994178469034 ± 5.2e-04, 0.4323202059145284 ± 4.3e-04, 0.5453189821025777 ± 5.5e-04, 0.5230627838972886 ± 5.2e-04, 0.4727933968885517 ± 4.7e-04]])
E             comparison failed. Mismatched elements: 97 / 99:
E             Max absolute difference: 0.023020796747366568
E             Max relative difference: 0.0515452071888622
E             Index   | Obtained            | Expected                     
E             (0, 0)  | 0.5661895453802076  | 0.5749909668197436 ± 5.7e-04 
E             (0, 1)  | 0.4897482151196382  | 0.4982809318565699 ± 5.0e-04 
E             (0, 2)  | 0.31553973402864666 | 0.3134150250073149 ± 3.1e-04 
E             (0, 3)  | 0.5162090524409638  | 0.5207541150644102 ± 5.2e-04 
E             (0, 4)  | 0.49924361453446325 | 0.5113092316865084 ± 5.1e-04 
E             (0, 5)  | 0.4346309015492143  | 0.44782135221718467 ± 4.5e-04
E             (0, 6)  | 0.5580530000456343  | 0.5638358511267394 ± 5.6e-04 
E             (0, 7)  | 0.494283055883771   | 0.5087362501516259 ± 5.1e-04 
E             (0, 8)  | 0.4536075655061833  | 0.4583123800154211 ± 4.6e-04 
E             (1, 0)  | 0.40089888366512905 | 0.40391136094126817 ± 4.0e-04
E             (1, 1)  | 0.44892064516173624 | 0.4629188577066109 ± 4.6e-04 
E             (1, 2)  | 0.32447588152438783 | 0.3270217988289004 ± 3.3e-04 
E             (1, 3)  | 0.47145566900785074 | 0.4744905988910448 ± 4.7e-04 
E             (1, 4)  | 0.4440828233775116  | 0.45122069617141375 ± 4.5e-04
E             (1, 5)  | 0.3505000307702088  | 0.36473938291538227 ± 3.6e-04
E             (1, 6)  | 0.4428473154873707  | 0.45076334272894086 ± 4.5e-04
E             (1, 7)  | 0.43704641084127316 | 0.45627088548274264 ± 4.6e-04
E             (1, 8)  | 0.45059925088255187 | 0.45724005049219124 ± 4.6e-04
E             (2, 0)  | 0.4727985335706301  | 0.49234843880437607 ± 4.9e-04
E             (2, 1)  | 0.47122498272003693 | 0.4837867784820462 ± 4.8e-04 
E             (2, 2)  | 0.299678225660559   | 0.30479843354026726 ± 3.0e-04
E             (2, 3)  | 0.38130872259859594 | 0.3877904779637883 ± 3.9e-04 
E             (2, 4)  | 0.48611073020253115 | 0.5091315269498977 ± 5.1e-04 
E             (2, 5)  | 0.3933841675807132  | 0.41258599322508044 ± 4.1e-04
E             (2, 6)  | 0.48564241723881785 | 0.5073916624412993 ± 5.1e-04 
E             (2, 7)  | 0.4220891807047905  | 0.4335772634543771 ± 4.3e-04 
E             (2, 8)  | 0.4099690487116389  | 0.41889222577946744 ± 4.2e-04
E             (3, 0)  | 0.5448322431311445  | 0.5581575375417775 ± 5.6e-04 
E             (3, 1)  | 0.48775067242321607 | 0.4956813043068013 ± 5.0e-04 
E             (3, 2)  | 0.25332227683264197 | 0.24939335696155615 ± 2.5e-04
E             (3, 3)  | 0.4683176368687046  | 0.4705759379193394 ± 4.7e-04 
E             (3, 4)  | 0.46327430098169964 | 0.4675657676343013 ± 4.7e-04 
E             (3, 5)  | 0.37386518438129823 | 0.38311515208298363 ± 3.8e-04
E             (3, 6)  | 0.48512387643464566 | 0.49676797773027376 ± 5.0e-04
E             (3, 7)  | 0.41554466894601644 | 0.4205948863686782 ± 4.2e-04 
E             (3, 8)  | 0.3807226752658629  | 0.38195189904435023 ± 3.8e-04
E             (4, 0)  | 0.425955570352543   | 0.4447724991306245 ± 4.4e-04 
E             (4, 1)  | 0.3684770892451089  | 0.3783054208340637 ± 3.8e-04 
E             (4, 2)  | 0.4410881267753337  | 0.4390701786134871 ± 4.4e-04 
E             (4, 3)  | 0.36694191113560315 | 0.36846865244987703 ± 3.7e-04
E             (4, 4)  | 0.5252871690349405  | 0.5270471107170016 ± 5.3e-04 
E             (4, 5)  | 0.5132101896548009  | 0.5245109429718812 ± 5.2e-04 
E             (4, 6)  | 0.4689208190686408  | 0.48478513819496744 ± 4.8e-04
E             (4, 7)  | 0.5065057186878028  | 0.5166881255308146 ± 5.2e-04 
E             (4, 8)  | 0.44629376912457536 | 0.4603307499137023 ± 4.6e-04 
E             (5, 0)  | 0.47735513434041116 | 0.48662240966122744 ± 4.9e-04
E             (5, 1)  | 0.4689783082018831  | 0.48060577303582214 ± 4.8e-04
E             (5, 2)  | 0.3204358784200522  | 0.318920320846488 ± 3.2e-04  
E             (5, 3)  | 0.4408106227783075  | 0.4402181698608169 ± 4.4e-04 
E             (5, 4)  | 0.5007626818807429  | 0.5065287479959606 ± 5.1e-04 
E             (5, 5)  | 0.3699317238093036  | 0.3808428016344234 ± 3.8e-04 
E             (5, 6)  | 0.4878948747389255  | 0.503760518497169 ± 5.0e-04  
E             (5, 7)  | 0.5045809752271339  | 0.5119492939077402 ± 5.1e-04 
E             (6, 0)  | 0.49088619417352036 | 0.4916554733557071 ± 4.9e-04 
E             (6, 1)  | 0.3242598312127636  | 0.32723491697559476 ± 3.3e-04
E             (6, 2)  | 0.2516160205096413  | 0.2503988728840708 ± 2.5e-04 
E             (6, 3)  | 0.31060926425620045 | 0.3060085611576825 ± 3.1e-04 
E             (6, 4)  | 0.32683950859193184 | 0.33358102847477644 ± 3.3e-04
E             (6, 5)  | 0.3790449278615964  | 0.38624049162552265 ± 3.9e-04
E             (6, 6)  | 0.4389288752285732  | 0.4409724167911789 ± 4.4e-04 
E             (6, 7)  | 0.3162662837329194  | 0.3228061546269177 ± 3.2e-04 
E             (6, 8)  | 0.29035103345227875 | 0.2965345226568545 ± 3.0e-04 
E             (7, 0)  | 0.2871659081637656  | 0.2845018638550871 ± 2.8e-04 
E             (7, 1)  | 0.28664174471968445 | 0.28188355320070757 ± 2.8e-04
E             (7, 2)  | 0.36327047305945614 | 0.3746961222903527 ± 3.7e-04 
E             (7, 3)  | 0.24486393627597738 | 0.239383289320872 ± 2.4e-04  
E             (7, 4)  | 0.3293723861800157  | 0.3380651566374858 ± 3.4e-04 
E             (7, 5)  | 0.328144844263206   | 0.33980877688422756 ± 3.4e-04
E             (7, 7)  | 0.3161032834338174  | 0.32213977069801136 ± 3.2e-04
E             (7, 8)  | 0.3850523973750647  | 0.3963803850825761 ± 4.0e-04 
E             (8, 0)  | 0.4954660515399947  | 0.5094956203659733 ± 5.1e-04 
E             (8, 1)  | 0.3909431003410763  | 0.40490180684472027 ± 4.0e-04
E             (8, 2)  | 0.36189466084216904 | 0.3638743592115522 ± 3.6e-04 
E             (8, 3)  | 0.3854066017072128  | 0.3839958751874655 ± 3.8e-04 
E             (8, 4)  | 0.4884735045393706  | 0.5032914529263811 ± 5.0e-04 
E             (8, 5)  | 0.47326652667330954 | 0.4885345815871985 ± 4.9e-04 
E             (8, 6)  | 0.5097694004001     | 0.5174383521666145 ± 5.2e-04 
E             (8, 7)  | 0.39110273589139    | 0.40837256558614843 ± 4.1e-04
E             (8, 8)  | 0.38815714379803656 | 0.40339167731136283 ± 4.0e-04
E             (9, 0)  | 0.35266545025595114 | 0.3510676862025384 ± 3.5e-04 
E             (9, 1)  | 0.3268139700129895  | 0.3244424797642443 ± 3.2e-04 
E             (9, 2)  | 0.41884707579916236 | 0.4328676030379516 ± 4.3e-04 
E             (9, 3)  | 0.27567100332903394 | 0.27242857987215746 ± 2.7e-04
E             (9, 4)  | 0.4014720849553264  | 0.42216604675489317 ± 4.2e-04
E             (9, 5)  | 0.39980973770465217 | 0.41882884230174644 ± 4.2e-04
E             (9, 6)  | 0.4179075963326111  | 0.4169958992359919 ± 4.2e-04 
E             (9, 7)  | 0.3815216796312004  | 0.3957578636483203 ± 4.0e-04 
E             (9, 8)  | 0.45740623248482026 | 0.4698541251673909 ± 4.7e-04 
E             (10, 0) | 0.594005385050121   | 0.6039527453563808 ± 6.0e-04 
E             (10, 1) | 0.5194816094414755  | 0.5220245309404515 ± 5.2e-04 
E             (10, 2) | 0.30762344430739896 | 0.30428597130705143 ± 3.0e-04
E             (10, 3) | 0.4664958099180245  | 0.4691870470253766 ± 4.7e-04 
E             (10, 4) | 0.49927177352388774 | 0.5168994178469034 ± 5.2e-04 
E             (10, 5) | 0.4232217596999662  | 0.4323202059145284 ± 4.3e-04 
E             (10, 6) | 0.5395768524251815  | 0.5453189821025777 ± 5.5e-04 
E             (10, 7) | 0.5144397329395007  | 0.5230627838972886 ± 5.2e-04 
E             (10, 8) | 0.46701999626091434 | 0.4727933968885517 ± 4.7e-04

../../../src/python/test/test_representations.py:104: AssertionError
```

Reply:
There are 2 versions of the code, in _sliced_wasserstein_distance and _compute_persistence_diagram_projections, if one is modified the other should be as well I think.